### ISSUE TITLE
[3.4] Using a contenttype with dash throws exception, instead of the notice to fix it on the dashboard.

### ISSUE THREAD

**Issue by bobdenotter** [bolt/bolt#7123]
If you have a contenttype like this: 

```yml
dummies-foo:
    slug: dummies-foo
    name: Dummies
    singular_name: Dummy
    singular-slug: dummy-foo
    fields:
        title:
            type: text
            class: large
            group: content
        slug:
            type: slug
            uses: title
        teaser:
            type: html
            height: 150px
```

The dashboard throws an exception: 

![screen shot 2017-10-26 at 12 22 19](https://user-images.githubusercontent.com/1833361/32048051-57c0cb44-ba48-11e7-9db4-bd26ccc38dd6.png)

In 3.3 it showed the notice how to fix it: 

![screen shot 2017-10-26 at 12 25 12](https://user-images.githubusercontent.com/1833361/32048148-c55db342-ba48-11e7-9d30-62a5e029a8c8.png)

**Comment by bobdenotter** [comment]
Two potential fixes come to mind, in https://github.com/bolt/bolt/blob/3.4/src/Config.php#L885 :

1) ignore the contenttype, until it's fixed: 

```php
            if (strpos($key, '-') !== false) {
                $error = Trans::__(
                    'contenttypes.generic.invalid-hyphen',
                    [
                        '%contenttype%' => $key,
                    ]
                );
                $this->app['logger.flash']->error($error);
                unset($this->data['contenttypes'][$key]);

                $passed = false;
            }
```

This will effectively ignore the contenttype, and jsut display the notice.

2) "Fix" it, and move on:

```php
            if (strpos($key, '-') !== false) {
                $original = $this->data['contenttypes'][$key];
                unset($this->data['contenttypes'][$key]);
                $key = str_replace('-', '_', strtolower(Str::makeSafe($key, true)));
                $this->data['contenttypes'][$key] = $original;
            }
```

This will treat it, as if it were `dummies_foo`, and at first glance all seems to work well.

**Comment by bobdenotter** [comment]
Ok, option 2 is a can of worms. I'll PR a fix with option 1.

**Comment by develth** [comment]
HI @bobdenotter , 

when using a taxonomy with a - it also throws exceptions ( after clearing cache there are db select exceptions ). 
Could this be releted, or should i open a new issue?

Cheers, 
Thomas

**Comment by bobdenotter** [comment]
Hi, @develth,

It probably is related, and we do want to fix that (or at least show a user friendly message). Feel free to open an issue, and I can take a look tomorrow! :-)

**Comment by bobdenotter** [comment]
Fixed in #7125
